src/linear_algebra/vectors.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2021 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Kyle Miller
+-/
+
+import data.matrix.basic
+import data.matrix.notation
+import tactic.fin_cases
+
+/-!
+# Vectors
+
+This module provides lemmata to more easily manipulate explicit vectors in $R^2$ and $R^3$
+for $R$ a commutative ring.
+
+## Notation
+
+The locale `vectors` gives the following notation:
+
+* `⬝` for dot products
+
+## Tags
+
+vectors
+-/
+
+variable {R : Type*}
+
+lemma vec2_eq {a₀ a₁ b₀ b₁ : R} (h₀ : a₀ = b₀) (h₁ : a₁ = b₁) :
+  ![a₀, a₁] = ![b₀, b₁] :=
+by { ext x, fin_cases x; assumption }
+
+lemma vec3_eq {a₀ a₁ a₂ b₀ b₁ b₂ : R} (h₀ : a₀ = b₀) (h₁ : a₁ = b₁) (h₂ : a₂ = b₂) :
+  ![a₀, a₁, a₂] = ![b₀, b₁, b₂] :=
+by { ext x, fin_cases x; assumption }
+
+variable [comm_ring R]
+
+lemma vec2_add {a₀ a₁ b₀ b₁ : R} :
+  ![a₀, a₁] + ![b₀, b₁] = ![a₀ + b₀, a₁ + b₁] :=
+by { ext x, fin_cases x; refl }
```

Comment:
I would argue this should be proved more generally as a `cons_add_cons` lemma next to `add_cons` in `fin/vec_notation`. The proof of this then becomes `rw [cons_add_cons, cons_add_cons, empty_add_empty]`

Reply:
I've gone ahead and added that lemma in #11241, to see if anything breaks.

Reply:
So you want to have both the general version and the 2D version, don't you? And of course, the 3D version will be preserved.

Reply:
@madvorak I think what Eric means is that rather than proving this by `fin_cases`, he would prefer it being proved using a some rewrites. The 3D version I guess would have the proof `rw [cons_add_cons, cons_add_cons, cons_add_cons, empty_add_empty]`.  (Or maybe both will just be `by simp`!)

Reply:
With all these issues, I will wait for https://github.com/leanprover-community/mathlib/pull/11241 to be merged before I make any changes to what currently is in `linear_algebra/vectors` ok?